src/parser/jobs/drg/modules/Positionals.js

```diff
@@ -0,0 +1,244 @@
+			this._tnCharges += 1
+
+			if (this._tnCharges < TRUE_NORTH_CHARGES) {
+				// if we're below the max, queue another charge since the cd will keep ticking
+				this._nextChargeIn += TRUE_NORTH_CD
+			} else {
+				// otherwise we're good, no recharge happening
+				this._nextChargeIn = null
+			}
+		}
+
+		// if we're over the max, uhhhhhhhhhh panic?
+		if (this._tnCharges > TRUE_NORTH_CHARGES) {
+			console.log('DRG Error: True north charges exceeded limit')
+		}
+	}
+
+	_tnUsed(event) {
+		// update
+		this._updateTnCharges(event.timestamp)
+
+		// remove charge
+		this._tnCharges -= 1
+
+		// mark next recharge time, if we don't have one going already
+		if (!this._nextChargeIn) {
+			this._nextChargeIn = event.timestamp + TRUE_NORTH_CD
+		}
+
+		// if we're below 0, panic
+		if (this._tnCharges < 0) {
+			console.log('DRG Error: True north charges went below 0')
+		}
+	}
+
+	_procSuccess() {
+		// you did it!
+		if (this._currentCombo) {
+			this._currentCombo.success = true
+		}
+
+		// don't actually close until next gcd happens
+	}
+
+	_onComplete() {
+		const missed = this._rtCombos.filter(combo => !combo.success).length
+
+		this.suggestions.add(
+			new TieredSuggestion({
+				icon: ACTIONS.RAIDEN_THRUST.icon,
+				content: <Trans id="drg.positionals.suggestions.content">Performing the proper positionals for	<ActionLink {...ACTIONS.FANG_AND_CLAW} /> and <ActionLink {...ACTIONS.WHEELING_THRUST} /> will do more damage and	let you use <ActionLink {...ACTIONS.RAIDEN_THRUST} />. Try to land these positionals, and use <ActionLink {...ACTIONS.TRUE_NORTH} /> if	it's available.	</Trans>,
+				tiers: {
+					1: SEVERITY.MINOR,
+					10: SEVERITY.MEDIUM,
+				},
+				value: missed,
+				why: <Trans id="drg.positionals.suggestions.why">You missed {missed} <Plural value={missed} one="proc" other="procs" /> of Raiden Thrust.</Trans>,
+			})
+		)
+	}
+
+	_checkIcon(success, size = 'large') {
+		return success ? <Icon color="green" name="check" size={size} /> : <Icon color="red" name="x" size={size} />
+	}
+
+	_procTable() {
+		return this._rtCombos.map(combo => {
+			const action = getDataBy(ACTIONS, 'id', combo.next)
+
+			return <Table.Row key={combo.time}>
+				<Table.Cell>{this.jumps.createTimelineButton(combo.time)}</Table.Cell>
+				<Table.Cell><ActionLink {...action} /></Table.Cell>
+				<Table.Cell	textAlign="center" positive={combo.success}	negative={!combo.success}>
+					{this._checkIcon(combo.success)}
+				</Table.Cell>
+				<Table.Cell textAlign="center">
+					{combo.trueNorthCast1 ? <Icon color="green" name="check" /> : ''}
+				</Table.Cell>
+				<Table.Cell textAlign="center">
+					{combo.trueNorthCast2 ? <Icon color="green" name="check" /> : ''}
+				</Table.Cell>
+				<Table.Cell textAlign="center">
+					{this._checkIcon(combo.trueNorthCharges > 0, '')} ({combo.trueNorthCharges} <Plural	value={combo.trueNorthCharges} one="charge" other="charges"	/>)
+				</Table.Cell>
+			</Table.Row>
+		})
+	}
+
+	output() {
```

Comment:
I'm happy to debate this one, but I don't think adding an output table for Raiden Thrust will provide as much value as it does for the other modules where we currently do it, mostly because of the oversaturation of data. Raiden Thrust (ideally) gets used every rotation, which should be something like 23s long, give or take, so you're going to have 25-30 rows for an average fight. I'm going to pull down this branch later and see how it actually looks just in case that changes my mind, but it's worth considering the risk of overwhelming a user with too much information.

Reply:
I ended up debating this for a while myself. I put the full table there just to see how it looked, but it can be easily filtered down to missed procs only. It is kinda interesting to see when and where TN ended up being used over the course of the fight, but is indeed quite a bit of info.